psbt/creator.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2018 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package psbt
+
+// The Creator has only one function:
+// takes a list of inputs and outputs and converts them to
+// the simplest Psbt, i.e. one with no data appended to inputs or outputs,
+// but only the raw unsigned transaction in the global section.
+
+import (
+	"github.com/btcsuite/btcd/wire"
+)
+
+// Creator holds a reference to a created Psbt struct.
+type Creator struct {
+	Cpsbt *Psbt
+}
+
+// createPsbt, on provision of an input and output 'skeleton' for
+// the transaction, returns a Creator struct.
+// Note that we require OutPoints and not TxIn structs, as we will
+// only populate the txid:n information, *not* any scriptSig/witness
+// information. The values of nLockTime, nSequence (per input) and
+// transaction version (must be 1 of 2) must be specified here. Note
+// that the default nSequence value is wire.MaxTxInSequenceNum.
+func (c *Creator) createPsbt(inputs []*wire.OutPoint,
```

Comment:
Shouldn't this method be exported so that code outside of package can create a psbt?

Reply:
Hmm, yes! Strange oversight; there's NewPsbtFromUnsignedTx and NewPsbt (which reads in a serialization instead of creating from scratch) exported in the main psbt.go but for some reason I overlooked that this method ought to be exported too. Come to think of it, the other two methods should perhaps be more logically in this file also.

Reply:
Done in cee0fa461626d53d87dc3f0b8880af05e9bfbd53